src/net_processing.cpp

```diff
@@ -3927,17 +3927,16 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
         if (pto->IsAddrRelayPeer() && pto->m_next_addr_send < current_time) {
             pto->m_next_addr_send = PoissonNextSend(current_time, AVG_ADDRESS_BROADCAST_INTERVAL);
             std::vector<CAddress> vAddr;
-            vAddr.reserve(pto->vAddrToSend.size());
+            vAddr.reserve(std::min(pto->vAddrToSend.size(), (size_t)MAX_ADDR_TO_SEND));
```

Comment:
style-nit: Instead of c-style casting of one or both arguments, selecting the exact template will force the type of both arguments and the return type as well. (Just casting one argument could still yield a different return type otherwise). Though, please don't change if there is nothing else to change

```suggestion
            vAddr.reserve(std::min<size_t>(pto->vAddrToSend.size(), MAX_ADDR_TO_SEND));
```

Reply:
Or change the type of `MAX_ADDR_TO_SEND` from `unsigned int` to `size_t` (`std::vector::size()` always returns `size_t`).

Reply:
> Or change the type of `MAX_ADDR_TO_SEND` from `unsigned int` to `size_t` (`std::vector::size()` always returns `size_t`).

To be pedantic, `std::vector::size()` always returns `size_type` which [is](https://en.cppreference.com/w/cpp/container/vector)
>  Unsigned integer type (_usually_ std::size_t) 

:)

Reply:
:ok_hand: :bulb: